src/game-engine/world/action/prayer.action.ts

```diff
@@ -0,0 +1,75 @@
+import { Player } from '@engine/world/actor/player/player';
+import { ActionHook, getActionHooks } from '@engine/world/action/hooks';
+import { advancedNumberHookFilter, questHookFilter } from '@engine/world/action/hooks/hook-filters';
+import { ActionPipe, RunnableHooks } from '@engine/world/action/index';
+import { Npc } from '../actor/npc/npc';
+
+
+/**
+ * Defines a button action hook.
+ */
+export interface PrayerActionHook extends ActionHook<PrayerAction, PrayerActionHandler> {
+    // The npc world id that was clicked on after choosing the spell

```

Comment:
i handles button presses but isn't actually a magic attack - so it will follow the same format - but it still needs to be an action hook i think?

Reply:
The code in it just refers to NPC's after choosing a spell which doesn't really make sense I think? So I assumed it was also a copy from another action hook to later be changed to an actual prayer hook.

Prayer is always applied to the actor or enemy player actor in case of "Smite" and not another npc actor.

Reply:
Nvm my last sentence in my previous comment. Retribution is able to deal damage to nearby actors